### PR TITLE
Quest log in var db/quest-log

### PR DESCRIPTION
**Pull request by omattornot**
I add the var for new quest-log of letter, orum and fur quest.

### CODE REVIEW

**Comment by mekolat** [comment]
the bitmasking seems wrong..

**Comment by wushin** [comment]
Aye, @omattornot you need to replace the VAR name in the quest with the new var. Look at the other quests or chat with us in IRC.

File: world/map/db/quest-log.txt [hunk 1]
```diff
@@ -28,3 +28,10 @@
 20,QL_TERRANITE_ARMOR,QUEST_NorthTulimshar,6,4
 // 21 => assassin quest
 22,QL_CELESTIA,QUEST_Hurnscald,2,8
+23,QL_Caretaker,QUEST_Graveyard_Caretaker,0,4
```
Comment (mekolat) [inline world/map/db/quest-log.txt]: All `QL_` variables should be in ALL CAPS

- - -
`QUEST_Graveyard_Caretaker` should be moved to an existing bitmasked var, and converted in `clear_vars`. It only needs 3 bits so we can fit it easily somewhere

File: world/map/db/quest-log.txt [hunk 2]
```diff
@@ -28,3 +28,10 @@
...
+24,QL_Orum,OrumQuest,0,8
```
Comment (mekolat) [inline world/map/db/quest-log.txt]: only needs 6 bits

File: world/map/db/quest-log.txt [hunk 3]
```diff
@@ -28,3 +28,10 @@
...
+25,QL_Furquest,QUEST_WG_state,0,8
```
Comment (mekolat) [inline world/map/db/quest-log.txt]: this one could be reduced to only 5 bits

File: world/map/npc/002-2/troupe_leader.txt [hunk 4]
```diff
@@ -2,10 +2,10 @@
 
 002-2,81,21,0|script|Troupe Leader|165
 {
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/002-2/troupe_leader.txt]: Don't!
remove this line

File: world/map/npc/008-1/hinnak.txt [hunk 5]
```diff
@@ -7,16 +7,23 @@
 
 008-1,101,30,0|script|Hinnak|142
 {
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/008-1/hinnak.txt]: remove this line

File: world/map/npc/008-1/hinnak.txt [hunk 6]
```diff
@@ -7,16 +7,23 @@
...
 
-    if (@inspector == 10) goto L_NohMask;
+    if ((GM >= DBG_HINNAK || debug) && getequipid(equip_head) == 647)
```
Comment (mekolat) [inline world/map/npc/008-1/hinnak.txt]: :+1: nicely done

File: world/map/npc/008-1/hinnak.txt [hunk 7]
```diff
@@ -7,16 +7,23 @@
...
+    if (QL_Inspector == 10) goto L_NohMask;
     goto L_No_NohMask;
 
+L_DeveloperBoard:
+    callfunc "HinnakDebug";
+    close;
+
 L_No_NohMask:
-    set @state, ((QUEST_Hurnscald & NIBBLE_1_MASK) >> NIBBLE_1_SHIFT);
+    set QL_Hinnak, ((QUEST_Hurnscald & NIBBLE_1_MASK) >> NIBBLE_1_SHIFT);
```
Comment (mekolat) [inline world/map/npc/008-1/hinnak.txt]: remove this line

File: world/map/npc/008-1/hinnak.txt [hunk 8]
```diff
@@ -247,11 +252,37 @@ L_NohMask_Satchel:
...
-    set QUEST_Hurnscald, (QUEST_Hurnscald & ~(NIBBLE_1_MASK)) | (@state << NIBBLE_1_SHIFT);
-    return;
+}
+
+function|script|HinnakDebug
+{
+    goto L_Menu;
+
+L_Menu:
+    menu
```
Comment (mekolat) [inline world/map/npc/008-1/hinnak.txt]: this debug should by show the state in `L_Menu`, and put a `if (debug)` condition on `menu` so you can't change the quest var on main server

File: world/map/npc/009-1/jack.txt [hunk 9]
```diff
@@ -29,7 +29,7 @@
     set @Q_Forestbow, ((QUEST_Forestbow_state & @Q_Forestbow_MASK) >> @Q_Forestbow_SHIFT);
     set @Q_Woodenshield, ((QUEST_Forestbow_state & @Q_Woodenshield_MASK) >> @Q_Woodenshield_SHIFT);
 
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/009-1/jack.txt]: ...remove this line

File: world/map/npc/009-1/milly.txt [hunk 10]
```diff
@@ -8,7 +8,7 @@
 
 009-1,81,41,0|script|Milly|114
 {
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/009-1/milly.txt]: remove this line

File: world/map/npc/009-1/old_man.txt [hunk 11]
```diff
@@ -2,13 +2,13 @@
 
 009-1,85,51,0|script|Old Man|159
 {
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/009-1/old_man.txt]: remove

File: world/map/npc/009-1/old_man.txt [hunk 12]
```diff
@@ -30,11 +30,10 @@ L_Accuse:
 L_Accuse_Respond:
     mes "[Old Man]";
     mes "\"Yes, I hung around the theater a lot. I was an actor when I was younger. But I wasn't there that night. Me and the wife were at home all night.\"";
-    set @inspector, 7;
-    set QUEST_Hurnscald, (QUEST_Hurnscald & ~(NIBBLE_3_MASK)) | (@inspector << NIBBLE_3_SHIFT);
+    set QL_Inspector, 7;
+    set QUEST_Hurnscald, (QUEST_Hurnscald & ~(NIBBLE_3_MASK)) | (QL_Inspector << NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/009-1/old_man.txt]: remove this line

File: world/map/npc/009-1/old_woman.txt [hunk 13]
```diff
@@ -2,31 +2,31 @@
 
 009-1,29,43,0|script|Old Woman|154
 {
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/009-1/old_woman.txt]: remove

File: world/map/npc/009-1/sabine.txt [hunk 14]
```diff
@@ -2,12 +2,12 @@
 
 009-1,88,27,0|script|Sabine|106
 {
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/009-1/sabine.txt]: remove

File: world/map/npc/009-2/alan.txt [hunk 15]
```diff
@@ -17,7 +17,7 @@
 
     set @BROKENLOG_EXP, 15;
 
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/009-2/alan.txt]: remove

File: world/map/npc/009-2/doctor.txt [hunk 16]
```diff
@@ -2,7 +2,7 @@
 
 009-2,148,25,0|script|Doctor|107
 {
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/009-2/doctor.txt]: remove

File: world/map/npc/009-2/inspector.txt [hunk 17]
```diff
@@ -2,24 +2,24 @@
...
-    if (@inspector == 7) goto L_Alibi;
-    if (@inspector == 8) goto L_Alibi_2;
-    if (@inspector == 9) goto L_Alibi_3;
-    if (@inspector == 10) goto L_Satchel;
-    if (@inspector == 11) goto L_Basement;
-    if (@inspector == 12) goto L_Note;
-    if (@inspector == 13) goto L_TravelingTroupe_2;
-    if (@inspector == 14) goto L_Over;
-    if (@inspector == 15) goto L_End;
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/009-2/inspector.txt]: remove

File: world/map/npc/009-2/lena.txt [hunk 18]
```diff
@@ -3,14 +3,14 @@
...
 {
-    set @state, ((QUEST_Hurnscald & NIBBLE_0_MASK) >> NIBBLE_0_SHIFT);
-
-    if (@state >= 6) goto L_Lena_Done;
-    if (@state == 5) goto L_Lena_Success;
-    if (@state == 4) goto L_LBL_Fail;
-    if (@state == 3) goto L_LBL_Fail;
-    if (@state == 2) goto L_Lena_Bandit_Leader;
-    if (@state == 1) goto L_Lena_Fairy_Hat;
+    set QL_Bandit, ((QUEST_Hurnscald & NIBBLE_0_MASK) >> NIBBLE_0_SHIFT);
```
Comment (mekolat) [inline world/map/npc/009-2/lena.txt]: remove

File: world/map/npc/009-2/nurse.txt [hunk 19]
```diff
@@ -44,7 +44,7 @@
     set @Q_poison_SHIFT, NIBBLE_7_SHIFT;
     set @Q_poison, (QUEST_Forestbow_state & @Q_poison_MASK) >> @Q_poison_SHIFT;
 
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/009-2/nurse.txt]: remove

File: world/map/npc/011-4/bl_barrier.txt [hunk 20]
```diff
@@ -2,10 +2,10 @@
 
 011-4,111,121,0|script|#BL_Barrier|45,0,0
 {
-    set @state, ((QUEST_Hurnscald & NIBBLE_0_MASK) >> NIBBLE_0_SHIFT);
+    set @QL_Bandit, ((QUEST_Hurnscald & NIBBLE_0_MASK) >> NIBBLE_0_SHIFT);
```
Comment (mekolat) [inline world/map/npc/011-4/bl_barrier.txt]: remove this line

File: world/map/npc/011-4/bl_barrier.txt [hunk 21]
```diff
@@ -2,10 +2,10 @@
...
 
-    if (@state >= 6) goto L_Cave_Abandoned;
-    if (@state >= 4) goto L_Lord_Barrier_Open;
+    if (@QL_Bandit >= 6) goto L_Cave_Abandoned;
```
Comment (mekolat) [inline world/map/npc/011-4/bl_barrier.txt]: remove the `@`

File: world/map/npc/011-4/bl_barrier.txt [hunk 22]
```diff
@@ -2,10 +2,10 @@
...
+    if (@QL_Bandit >= 4) goto L_Lord_Barrier_Open;
```
Comment (mekolat) [inline world/map/npc/011-4/bl_barrier.txt]: remove the `@`

File: world/map/npc/011-6/barrier.txt [hunk 23]
```diff
@@ -1,8 +1,8 @@
 011-6,264,236,0|script|#BL_Barrier2|45,0,0
 {
-    set @state, ((QUEST_Hurnscald & NIBBLE_0_MASK) >> NIBBLE_0_SHIFT);
+    set @QL_Bandit, ((QUEST_Hurnscald & NIBBLE_0_MASK) >> NIBBLE_0_SHIFT);
```
Comment (mekolat) [inline world/map/npc/011-6/barrier.txt]: remove this line

File: world/map/npc/011-6/barrier.txt [hunk 24]
```diff
@@ -1,8 +1,8 @@
...
 
-    if (@state >= 6) goto L_Lord_Hideout_Open;
+    if (@QL_Bandit >= 6) goto L_Lord_Hideout_Open;
```
Comment (mekolat) [inline world/map/npc/011-6/barrier.txt]: remove the `@`

File: world/map/npc/011-6/bryant.txt [hunk 25]
```diff
@@ -3,11 +3,11 @@
 
 011-6,27,200,0|script|Bryant|126
 {
-    set @state, ((QUEST_Hurnscald & NIBBLE_0_MASK) >> NIBBLE_0_SHIFT);
+    set QL_Bandit, ((QUEST_Hurnscald & NIBBLE_0_MASK) >> NIBBLE_0_SHIFT);
```
Comment (mekolat) [inline world/map/npc/011-6/bryant.txt]: remove this line

File: world/map/npc/011-6/bryant.txt [hunk 26]
```diff
@@ -20,24 +20,21 @@ L_Bryant_Summon:
...
-    callsub S_Update_Var;
+    set QL_Bandit, 4;
     monster "011-6",260,250,"BanditLord",1065,1, "Bryant::OnBanditLordDead";
     goto L_Close;
 
 OnBanditLordDead:
     message strcharinfo(0), "You killed the bandit lord.";
-    set @state, ((QUEST_Hurnscald & NIBBLE_0_MASK) >> NIBBLE_0_SHIFT);
-    if (@state != 4)
+    set QL_Bandit, ((QUEST_Hurnscald & NIBBLE_0_MASK) >> NIBBLE_0_SHIFT);
```
Comment (mekolat) [inline world/map/npc/011-6/bryant.txt]: remove this line

File: world/map/npc/011-6/crastur.txt [hunk 27]
```diff
@@ -1,7 +1,7 @@
 011-6,172,250,0|script|Crastur the Ugly|418
 {
     set @spants_state, ((QUEST_WG_state & NIBBLE_2_MASK) >> NIBBLE_2_SHIFT);
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/011-6/crastur.txt]: remove

File: world/map/npc/013-3/barrier.txt [hunk 28]
```diff
@@ -2,10 +2,10 @@
 
 013-3,71,21,0|script|#DemonMineBarrier1#_M|32767,1,0
 {
-    set @state, ((QUEST_Hurnscald & NIBBLE_2_MASK) >> NIBBLE_2_SHIFT);
+    set QL_Demon_mask, ((QUEST_Hurnscald & NIBBLE_2_MASK) >> NIBBLE_2_SHIFT);
```
Comment (mekolat) [inline world/map/npc/013-3/barrier.txt]: remove

File: world/map/npc/013-3/mask_chest.txt [hunk 29]
```diff
@@ -2,10 +2,10 @@
 
 013-3,28,27,0|script|Chest#DemonMask|111
 {
-    set @state, ((QUEST_Hurnscald & NIBBLE_2_MASK) >> NIBBLE_2_SHIFT);
+    set QL_Demon_mask, ((QUEST_Hurnscald & NIBBLE_2_MASK) >> NIBBLE_2_SHIFT);
```
Comment (mekolat) [inline world/map/npc/013-3/mask_chest.txt]: remove

File: world/map/npc/018-2/angus.txt [hunk 30]
```diff
@@ -2,9 +2,9 @@
 
 018-2,31,39,0|script|Angus|147
 {
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/018-2/angus.txt]: remove

File: world/map/npc/018-2/books.txt [hunk 31]
```diff
@@ -2,9 +2,9 @@
 
 018-2,112,19,0|script|#Bookcase38|400,0,1
 {
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/018-2/books.txt]: remove

File: world/map/npc/018-2/books.txt [hunk 32]
```diff
@@ -13,11 +13,10 @@ L_NohMask:
     next;
     mes "The book has been hollowed out. Inside is a theater mask and a note that you cannot even begin to read.";
     next;
-    set @inspector, 12;
-    set QUEST_Hurnscald, (QUEST_Hurnscald & ~(NIBBLE_3_MASK)) | (@inspector << NIBBLE_3_SHIFT);
+    set QL_Inspector, 12;
+    set QUEST_Hurnscald, (QUEST_Hurnscald & ~(NIBBLE_3_MASK)) | (QL_Inspector << NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/018-2/books.txt]: remove

File: world/map/npc/018-2/caul.txt [hunk 33]
```diff
@@ -9,7 +9,7 @@
 
 018-2,37,22,0|script|Caul|107
 {
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/018-2/caul.txt]: remove

File: world/map/npc/018-2/miners.txt [hunk 34]
```diff
@@ -3,11 +3,11 @@
 
 018-2,85,23,0|script|Malek|109
 {
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/018-2/miners.txt]: remove

File: world/map/npc/018-2/miners.txt [hunk 35]
```diff
@@ -24,8 +24,8 @@ L_NohMask:
         "The inspector sent me here to investigate.", L_Investigation;
 
 L_Investigation:
-    set @inspector, 11;
-    set QUEST_Hurnscald, (QUEST_Hurnscald & ~(NIBBLE_3_MASK)) | (@inspector << NIBBLE_3_SHIFT);
+    set QL_Inspector, 11;
+    set QUEST_Hurnscald, (QUEST_Hurnscald & ~(NIBBLE_3_MASK)) | (QL_Inspector << NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/018-2/miners.txt]: remove

File: world/map/npc/018-2/receptionist.txt [hunk 36]
```diff
@@ -2,7 +2,7 @@
 
 018-2,50,24,0|script|Receptionist#mine|108
 {
-    set @inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
+    set QL_Inspector, ((QUEST_Hurnscald & NIBBLE_3_MASK) >> NIBBLE_3_SHIFT);
```
Comment (mekolat) [inline world/map/npc/018-2/receptionist.txt]: remove

File: world/map/npc/018-3/bookcase.txt [hunk 37]
```diff
@@ -2,9 +2,9 @@
 
 018-3,75,123,0|script|#DemonMineBookcase1|400
 {
-    set @state, ((QUEST_Hurnscald & NIBBLE_2_MASK) >> NIBBLE_2_SHIFT);
+    set QL_Demon_mask, ((QUEST_Hurnscald & NIBBLE_2_MASK) >> NIBBLE_2_SHIFT);
```
Comment (mekolat) [inline world/map/npc/018-3/bookcase.txt]: remove

File: world/map/npc/020-2/furquest.txt [hunk 38]
```diff
@@ -13,7 +13,7 @@
         "Nothing.", L_Close;
 
 L_ShowState:
-    set @wg_state, ((QUEST_WG_state & BYTE_0_MASK) >> BYTE_0_SHIFT);
+    set @wg_state, ((QL_Furquest & BYTE_0_MASK) >> BYTE_0_SHIFT);
```
Comment (mekolat) [inline world/map/npc/020-2/furquest.txt]: this file is all wrong, please remove `@wg_state` everywhere and replace it with `QL_Furquest`

File: world/map/npc/020-2/furquest.txt [hunk 39]
```diff
@@ -650,6 +650,6 @@ L_Close:
 }
 function|script|setWGState
```
Comment (mekolat) [inline world/map/npc/020-2/furquest.txt]: remove this function

File: world/map/db/quest-log.txt (hunk 2 above)
Comment (omattornot) [inline world/map/db/quest-log.txt]: For QL_ORUM, we can add him in QUEST_Hurnscald,4,6 ?

**Comment by wushin** [comment]
@omattornot you need to rebase. Let me know when you fix the conflict so I can restart test-server.

**Comment by omattornot** [comment]
@wushin I fixed the conflict. But what rebase i need to do ? Squash all commit in one ?

**Comment by wushin** [comment]
@omattornot the conflict was the rebasing.

**Comment by mekolat** [comment]
@omattornot is this done? if yes I'll review again. @wushin feel free to check too

**Comment by pazkero** [comment]
I'm not going to merge this, it needs a bit more than just a rebase at this point 😰 

I'm closing but if you want to fix and try again, please, be my guest. This is by no means a rejection, just spring cleaning.

**Pull request closed without merging**